README.md

```diff
@@ -277,28 +278,160 @@ This gives the start timestamps that can be used to correlate the events and the
+git clone https://github.com/0chain/0chain.git
+```
+
+Navigate to 0chain folder and run the script to build base docker image for unit testing :
+
+```
+cd 0chain
+./docker.local/bin/build.base.sh
+```
+
+The base image includes all the dependencies required to test the 0chain code.
+
+### Running Tests
+
+Now run the script containing unit tests .
+
+```
+./docker.local/bin/unit_test.sh 
+```
+
+The list of packages is optional, and if provided runs only the tests from those packages. Command for running unit tests with specific packages .
+
+```
+./docker.local/bin/unit_test.sh [<packages>]
 ```
 
-Then run the tests.
+###  Testing Steps
+
+Unit testing happens over a series of steps one after the other.
+
+#### Step 1: FROM zchain_build_base
+
+This `FROM`step does the required preparation and specifies the underlying OS architecture to use the build image. Here we are using the base image created in the build phase.
+
+#### Step 2: ENV SRC_DIR=/0chain
+
+ The SRC_DIR  variable is a reference to a filepath which contains the code from your pull request. Here `/0chain` directory is specified as it is the one which was cloned. 
+
+#### Step 3: Setting the `GO111Module` variable to `ON` 
+
+`GO111MODULE` is an environment variable that can be set when using `go` for changing how Go imports packages. It was introduced to help ensure a smooth transition to the module system. 
+
+`GO111MODULE=on` will force using Go modules even if the project is in your GOPATH. Requires `go.mod` to work.
 
+ Note: The default behavior in Go 1.16 is now **GO111MODULE**=on 
+
+#### Step 4: COPY ./code/go/0chain.net $SRC_DIR/go/0chain.net
+
+This step copies the code from the source path to the destination path.
+
+#### Step 5: RUN cd $SRC_DIR/go/0chain.net &&  go mod download
+
+The RUN command is an image build step which allows installing of application and packages requited for testing while the`go mod download` downloads the specific module versions you've specified in the `go.mod`file.
+
+#### Step 6: RUN cd $GOPATH/pkg/mod/github.com/valyala/gozstd@v1.5. &&     chmod -R +w . &&  make clean libzstd.a
+
+This step runs the gozstd package and provides write permissions to the directory. gozstd which is a go wrapper for zstd (library) provides Go bindings for the libzstd C library. The `make clean` is ran in the last to clean up the code and remove all the compiled object files from the source code
+
+#### Step 7: WORKDIR $SRC_DIR/go
+
+This step defines the working directory for running unit tests which is (0chain/code/go/0chain.net/).For all the running general unit tests their code coverage will be defined in the terminal like this
+
+```
+ok      0chain.net/chaincore/block      0.128s  coverage: 98.9% of statements
 ```
-$ ./docker.local/bin/unit_test.sh [<packages>]
+
+The above output shows 98.9% of code statements was covered with tests.
+
+Here is a sample output for all the unit test cases:
+
 ```
+?       0chain.net/chaincore    [no test files]
```

Comment:
Not convinced this output is useful here, Plus there is no way this output is going to be kept current.